great_expectations/data_context/store/ge_cloud_store_backend.py

```diff
@@ -522,11 +522,11 @@ def remove_key(self, key):
             )
 
     def _has_key(self, key: Tuple[str, ...]) -> bool:  # type: ignore[override]
-        # self.list_keys() generates a list of length 2 tuples
-        if len(key) == 3:
-            key = key[:2]
-        all_keys = self.list_keys()
-        return key in all_keys
+        # Due to list_keys being inconsistently sized (due to the possible of resource names),
+        # we solely rely on the equality of ids to determine the result of this method
+        id = key[1]
+        all_ids = set(map(lambda key: key[1], self.list_keys()))
+        return id in all_ids
```

Comment:
What are the possible tuple sizes (and values) here?

This is a good case for a `@overload`
https://mypy.readthedocs.io/en/stable/more_types.html?highlight=overload#function-overloading
I'm happy to create the overload.

Reply:
`key[0]` should be consistent but I don't think it hurts to compare (especially since we have slight deviations with things like contracts vs checkpoints)